src/app/domains/views/DomainDetails/ResourceRecords/ResourceRecords.tsx

```diff
@@ -154,32 +135,42 @@ const ResourceRecords = ({ id }: Props): React.ReactElement | null => {
         },
         {
           content: (
-            <ContextualMenu
-              hasToggleIcon={true}
-              links={[
-                {
-                  children: "Edit record...",
-                  onClick: () => {
-                    setExpanded({
-                      content: RecordActions.EDIT,
-                      id: rowId,
-                    });
+            <Tooltip message={!canEdit ? Labels.DisabledActions : null}>
```

Comment:
I think you should make the tooltip more granular, there are two conditions to editing.
```suggestion
            <Tooltip message={isAutogenerated ? "System-generated records cannot be edited" : isSuperUser.data ? "You do not have permissions to edit this record."}>
```